src/GeniusYield/GYConfig.hs

```diff
@@ -203,15 +205,25 @@ withCfgProviders
             )
 
       bracket (Katip.mkKatipLog ns cfgLogging) logCleanUp $ \gyLog' -> do
-        (gyQueryUTxO, gySlotActions) <-
-            {-if cfgUtxoCacheEnable
-            then do
-                (gyQueryUTxO, purgeCache) <- CachedQuery.makeCachedQueryUTxO gyQueryUTxO' gyLog'
-                -- waiting for the next block will purge the utxo cache.
-                let gySlotActions = gySlotActions' { gyWaitForNextBlock' = purgeCache >> gyWaitForNextBlock' gySlotActions'}
-                pure (gyQueryUTxO, gySlotActions)
-            else-} pure (gyQueryUTxO', gySlotActions')
-        e <- try $ f GYProviders {..}
+        (gyQueryUTxO, gySlotActions, fT) <-
+          {-if cfgUtxoCacheEnable
+          then do
+              (gyQueryUTxO, purgeCache) <- CachedQuery.makeCachedQueryUTxO gyQueryUTxO' gyLog'
+              -- waiting for the next block will purge the utxo cache.
+              let gySlotActions = gySlotActions' { gyWaitForNextBlock' = purgeCache >> gyWaitForNextBlock' gySlotActions'}
+              pure (gyQueryUTxO, gySlotActions, f)
+          else pure (gyQueryUTxO', gySlotActions', f) -}
+          if cfgLogTiming then do
+              let fT = \provider -> do
+                    start <- getCurrentTime
+                    a <- f provider
```

Comment:
Hm.. Need to think if this is appropriate. You are wrapping around general call to provider and not to call of individual query calls.

Reply:
Time logs are now added to individual query calls.